test/system/irohad_test.cpp

```diff
@@ -80,16 +105,49 @@ class IrohadTest : public AcceptanceFixture {
 
   std::string params(const boost::optional<std::string> &config_path,
                      const boost::optional<std::string> &genesis_block,
-                     const boost::optional<std::string> &keypair_path) {
+                     const boost::optional<std::string> &keypair_path,
+                     const boost::optional<std::string> &additional_params) {
     std::string res;
     config_path | [&res](auto &&s) { res += " --config " + s; };
     genesis_block | [&res](auto &&s) { res += " --genesis_block " + s; };
     keypair_path | [&res](auto &&s) { res += " --keypair_name " + s; };
+    additional_params | [&res](auto &&s) { res += " " + s; };
     return res;
   }
 
   std::string setDefaultParams() {
-    return params(config_copy_, path_genesis_.string(), path_keypair_.string());
+    return params(
+        config_copy_, path_genesis_.string(), path_keypair_.string(), {});
+  }
+
+  iroha::protocol::ToriiResponse sendDefaultTx(
+      boost::optional<shared_model::crypto::Keypair> key_pair) {
```

Comment:
You don't check, if this optional contains value inside the function. Consider either passing raw value or adding the check.
Also, I think, in both cases argument should be passed by const ref

Reply:
I also suggest passing keypair as `const shared_model::crypto::Keypair &key_pair` since we use `boost::optional` when the filed may be missing.